src/controllers/sound/SoundController.cpp

```diff
@@ -0,0 +1,195 @@
+
+    /// Initialize a sound device
+    auto deviceConfig = ma_device_config_init(ma_device_type_playback);
+    deviceConfig.playback.pDeviceID = nullptr;
+    deviceConfig.playback.format = this->resourceManager->config.decodedFormat;
+    deviceConfig.playback.channels = 0;
+    deviceConfig.pulse.pStreamNamePlayback = "Chatterino MA";
+    deviceConfig.sampleRate = this->resourceManager->config.decodedSampleRate;
+    deviceConfig.dataCallback = ma_engine_data_callback_internal;
+    deviceConfig.pUserData = this->engine.get();
+
+    result =
+        ma_device_init(this->context.get(), &deviceConfig, this->device.get());
+    if (result != MA_SUCCESS)
+    {
+        qCWarning(chatterinoSound) << "Error initializing device:" << result;
+        return;
+    }
+
+    result = ma_device_start(this->device.get());
+    if (result != MA_SUCCESS)
+    {
+        qCWarning(chatterinoSound) << "Error starting device:" << result;
+        return;
+    }
+
+    /// Initialize engine
+    auto engineConfig = ma_engine_config_init();
+    engineConfig.pResourceManager = this->resourceManager.get();
+    engineConfig.pDevice = this->device.get();
+    engineConfig.pContext = this->context.get();
+
+    result = ma_engine_init(&engineConfig, this->engine.get());
+    if (result != MA_SUCCESS)
+    {
+        qCWarning(chatterinoSound) << "Error initializing engine:" << result;
+        return;
+    }
+
+    /// Initialize default ping sounds
+    {
+        // TODO: Can we optimize this?
+        BenchmarkGuard b("init sounds");
+        for (auto i = 0; i < NUM_SOUNDS; ++i)
+        {
+            auto decoderConfig =
+                ma_decoder_config_init(ma_format_f32, 0, 48000);
+            // This must match the encoding format of our default ping sound
+            decoderConfig.encodingFormat = ma_encoding_format_wav;
+
+            auto dec = std::make_unique<ma_decoder>();
+            auto snd = std::make_unique<ma_sound>();
+
+            result = ma_decoder_init_memory(
+                (void *)this->defaultPingData.data(),
+                this->defaultPingData.size() * sizeof(char), &decoderConfig,
+                dec.get());
+
+            result = ma_sound_init_from_data_source(
+                this->engine.get(), dec.get(),
+                MA_RESOURCE_MANAGER_DATA_SOURCE_FLAG_DECODE, nullptr,
+                snd.get());
+
+            this->defaultPingDecoders.emplace_back(std::move(dec));
+            this->defaultPingSounds.emplace_back(std::move(snd));
+        }
+
+        qCDebug(chatterinoSound) << "init sound! ";
+    }
+
+    this->initialized = true;
+}
+
+SoundController::~SoundController()
+{
+    for (const auto &snd : this->defaultPingSounds)
+    {
+        ma_sound_uninit(snd.get());
+    }
+    this->defaultPingSounds.clear();
+    for (const auto &dec : this->defaultPingDecoders)
+    {
+        ma_decoder_uninit(dec.get());
+    }
+    this->defaultPingDecoders.clear();
```

Comment:
Remove clear

Reply:
Resolved in [`bed44d1` (#4334)](https://github.com/Chatterino/chatterino2/pull/4334/commits/bed44d18a311db365d0b706dc93ec32e3564f69f)